### PR TITLE
Fix compatibility with OpenCV 4.7.0

### PR DESCRIPTION
**Pull request by traversaro**
Fix https://github.com/ami-iit/bipedal-locomotion-framework/issues/586 .

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ All notable changes to this project are documented in this file.
 
 ### Fix
 - Return an invalid `PolyDriverDescriptor` if `description` is not found in `constructMultipleAnalogSensorsRemapper()` (https://github.com/ami-iit/bipedal-locomotion-framework/pull/569)
+- Fix compatibility with OpenCV 4.7.0 ()
```
Comment (S-Dafarra) [inline CHANGELOG.md:16]:
```suggestion
- Fix compatibility with OpenCV 4.7.0 (https://github.com/ami-iit/bipedal-locomotion-framework/pull/589)
```

**Review by S-Dafarra — APPROVED** [review]

**Comment by traversaro** [comment]
I guess also @prashanthr05 may be interested on this.

File: src/Perception/src/ArucoDetector.cpp [hunk 2]
```diff
@@ -55,6 +55,11 @@ class ArucoDetector::Impl
     /**
      * Utility map for choosing Aruco marker dictionary depending on user parameter
      */
+    // TODO(traversaro): when we drop support for OpenCV < 4.7.0, we can cleanup this part
+    // we can also check if there are other dictionary that should be added here
+#if (CV_VERSION_MAJOR >= 5) || (CV_VERSION_MAJOR == 4 && CV_VERSION_MINOR >= 7)
```
Comment (prashanthr05) [inline src/Perception/src/ArucoDetector.cpp:60]: Yes, there are certainly other dictionaries to be added here. I remember experiencing this issue earlier and purposefully leaving out some dictionaries from the lookup.

Apriltags are not added here. https://docs.opencv.org/3.4/dc/df7/dictionary_8hpp.html

**Review by GiulioRomualdi — APPROVED** [review]

**Comment by GiulioRomualdi** [comment]
Thank you @traversaro

**Pull request merged**